src/main/java/com/creativemd/littletiles/client/gui/SubGuiImport.java

```diff
@@ -48,15 +51,39 @@ public void onClicked(int x, int y, int button) {
                     NBTTagCompound nbt = JsonToNBT.getTagFromJson(textfield.text);
                     try {
                         LittleGridContext.get(nbt);
-                        sendPacketToServer(nbt);
+                        String name = canImport(LittlePreviews.getPreview(StructureStringUtils.importStructure(nbt), false));
+                        if (name.isEmpty())
+                            sendPacketToServer(nbt);
+                        else
+                            openButtonDialogDialog("Structure Type: " + name + " must be imported in creative mode.", "Ok");
                     } catch (RuntimeException e) {
                         openButtonDialogDialog("Invalid grid size " + nbt.getInteger("grid"), "Ok");
                     }
                 } catch (NBTException e) {
-                
+                    
                 }
             }
         });
     }
     
+    /** Gets the name of the structure type that is preventing the player from importing the structure in survival mode
+     * 
+     * @param previews
+     *            Previews that is looks through
+     * @return
+     *         The name of the structure that is preventing the player from importing */
+    private String canImport(LittlePreviews previews) {
+        LittleStructureType type = previews.getStructureType();
+        if (!type.canImport())
+            return type.id;
+        String typeName = "";
+        for (LittlePreviews child : previews.getChildren()) {
```

Comment:
Unfortunately this is not enough. What if the children have also children?

You could do something like:
```java
for (LittlePreviews child : previews.getChildren()) {
  typeName = canImport(child);
}
```
This should work no matter how many layers of children there are. Only downside is that only one structure is reported. This will only show up one error, but there might be more.